src/codemagic/tools/app_store_connect.py

```diff
@@ -148,6 +161,81 @@ def _delete_resource(self, resource_manager, resource_id: ResourceId, ignore_not
             else:
                 raise AppStoreConnectError(str(api_error))
 
+    @cli.action('list-builds',
+                BuildArgument.APPLICATION_ID_RESOURCE_ID_OPTIONAL,
+                BuildArgument.EXPIRED,
+                BuildArgument.NOT_EXPIRED,
+                BuildArgument.BUILD_ID_RESOURCE_ID,
+                BuildArgument.PRE_RELEASE_VERSION,
+                BuildArgument.PROCESSING_STATE,
+                BuildArgument.BUILD_VERSION_NUMBER)
+    def list_builds(self,
+                    application_id: Optional[ResourceId] = None,
+                    expired: Optional[bool] = None,
+                    not_expired: Optional[bool] = None,
+                    build_id: Optional[ResourceId] = None,
+                    pre_release_version: Optional[str] = None,
+                    processing_state: Optional[BuildProcessingState] = None,
+                    build_version_number: Optional[str] = None,
+                    should_print: bool = True) -> List[Build]:
+        """
+        List Builds from Apple Developer Portal matching given constraints
+        """
+        try:
+            expired_value = Argument.resolve_optional_two_way_switch(expired, not_expired)
+        except ValueError as ve:
+            flags = f'{BuildArgument.EXPIRED.flags[0]!r} and {BuildArgument.NOT_EXPIRED.flags[0]!r}'
+            raise BuildArgument.NOT_EXPIRED.raise_argument_error(f'Using mutually exclusive switches {flags}.')
+
+        builds_filter = self.api_client.builds.Filter(
+            app=application_id,
+            expired=expired_value,
+            id=build_id,
+            processing_state=processing_state,
+            version=build_version_number,
+            pre_release_version_version=pre_release_version,
+        )
+        return self._list_resources(builds_filter, self.api_client.builds, should_print)
+
+    def get_latest_build_number(self, builds: List[Build]) -> Optional[int]:
+        try:
+            latest_build_number = max((build.attributes.version for build in builds), key=int)
+        except ValueError:
+            return None
+        else:
+            self.echo(str(latest_build_number))
+            return latest_build_number
```

Comment:
`else` block is rendundant since `except` clause terminates method anyway. 
```suggestion
        self.echo(str(latest_build_number))
        return latest_build_number
```